### PR TITLE
fix Slack links with ] in them

### PR DESCRIPTION
**Pull request by r3m0t**

### CODE REVIEW

**Comment by zoni** [comment]
Thanks!

File: errbot/backends/slack.py [hunk 1]
```diff
@@ -61,7 +61,7 @@
     'cyan': '#00FFFF'
 }  # Slack doesn't know its colors
 
-MARKDOWN_LINK_REGEX = re.compile(r'(?<!!)\[(?P<text>.+?)\]\((?P<uri>[a-zA-Z0-9]+?:\S+?)\)')
+MARKDOWN_LINK_REGEX = re.compile(r'(?<!!)\[(?P<text>[^\]]+?\]\((?P<uri>[a-zA-Z0-9]+?:\S+?)\)')
```
Comment (zoni) [inline errbot/backends/slack.py]: Huh, yeah, that door look weird. I overlooked that...

Comment (gbin) [inline errbot/backends/slack.py]: Was this the intention ?
```
re.compile(r'(?<!!)\[(?P<text>[^\]+?)\]\((?P<uri>[a-zA-Z0-9]+?:\S+?)\)')
```

Comment (r3m0t) [inline errbot/backends/slack.py]:
```python
MARKDOWN_LINK_REGEX = re.compile(r'(?<!!)\[(?P<text>[^\]]+?)\]\((?P<uri>[a-zA-Z0-9]+?:\S+?)\)')```

**Comment by r3m0t** [comment]
I guess the tests don't run on Travis

**Comment by zoni** [comment]
> I guess the tests don't run on Travis

Indeed, these weren't. The naming of the test file caused `pytest` to not pick it up. I renamed it (`tests/backend_tests/{slack_tests.py => slack_test.py}`) so that it is now being run again which should avoid this in the future.

I also fixed up the regex in master.

**Pull request merged**